PyEMD/significancetest.py

```diff
@@ -0,0 +1,127 @@
+from PyEMD import EMD, EEMD, CEEMDAN
+import numpy as np
+from scipy import stats
+import math
+
+
+#helper function: normalize data
+def normalize(data):
+    mean = sum(data)/len(data)
+    dmin = min(data)
+    dmax = max(data)
+    ret = np.array([])
+    for val in data:
+        ret = np.append(ret, (val-mean)/(dmax-dmin))
+    return ret
+
+#helper function: find zero-crossings in IMF, used for finding mean period of IMF
+def sign_change(data):
+    l = len(data)
+    counter=0
+    for x in range(l-1):
+        if (data[x]>0 or data[x] == 0) and data[x+1] < 0:
+            counter+=1
+    return counter
+
+#helper function: Find mean period of an IMF
+def mean_period(data):
+    s = sign_change(data)
+    if s != 0:
+        return len(data)/s
+    else:
+        return len(data)
+
+#helper function: find energy of signal/IMF
+def energy(data):
+    return sum(pow(data, 2))
+
+#helper function: find IMF significance in 'a priori' test
+def significance_apriori(energy_density, T, N, alpha):
+    k = abs(stats.norm.ppf((1-alpha)/2))
+    if (energy_density <= (-math.log(T) + k*(math.sqrt(2/N)*math.exp(math.log(T)/2)))) and (energy_density >= (-math.log(T) - k*(math.sqrt(2/N)*math.exp(math.log(T)/2)))):
+        return False
+    else:
+        return True
+
+#helper function: find significance in 'a posteriori' test
+def significance_aposteriori(scaled_energy_density, T, N, alpha):
+    k = abs(stats.norm.ppf((1-alpha)/2))
+    if (scaled_energy_density <= (-math.log(T) + k*(math.sqrt(2/N)*math.exp(math.log(T)/2)))):
+        return False
+    else:
+        return True
+
+def whitenoisecheck(data: np.ndarray, method: str='EMD',  test: str='aposteriori', rescaling_imf: int=1, alpha: float= 0.95):
```

Comment:
I like the name but let's make it a bit more pythonic. How about `whitenoise_check`? Python standard is to separate words with underscore `_` (unless package or filenames). Also, as expressed in another comment, since this is a **check** someone has likely already performed decomposition and only wants to check that. For this we don't need to perform another decomposition, thus the data should be `IMFs`.

Maybe singature like so:
```
def whitenoise_check(imfs: np.ndarray,  test: str = 'aposteriori', rescaling_imf: int = 1, alpha: float = 0.95):
```

Any chance you could also describe what parameters are and expected? As an example for doc please see here (https://github.com/laszukdawid/PyEMD/blob/master/PyEMD/utils.py).

Even though you have provided great reference(!) and example they would break documentation builder. For documentation we are using `Sphinx` which expects [RST markup language](https://docutils.sourceforge.io/rst.html). In particular it means that headings require same number of underlying characters, i.e.

```
References
----------

Examples
--------
```

Reply:
- `whitenoise_check` sounds wonderful. Its good to stick to standards, makes life a lot easier while debugging. Regarding the filename being `checks.py`, I am ok with it, but we have to make sure that in the documentation we make it explicit that it is the _'statistical significance test'_ else many might not know what it does.

- However, as mentioned above, the input data as IMFs may not be feasible, please see the detailed reason and the solution suggested in the comment https://github.com/laszukdawid/PyEMD/pull/105#discussion_r757439314

- I will surely add the necessary parameters in the next push and for documentation I am not exactly aware of RST markup language but will try to make the changes as suggested. Do give them a check once I push the changes.

Reply:
So far there's only one *statistical significance test* and it explicitly states that it's a statistical significance test. I think we'll be fine.